examples/tour.jl

```diff
@@ -0,0 +1,192 @@
+
+# Here, we return and load a single batch item
+# Note that the labels are not necessarily sampled at the same resolution as the signal.
+# This is because the labels are sampled at 0.033 Hz (1 every 30 seconds) while the signal
+# is sampled at 128 Hz.
+x, y = materialize_batch_item(item)
+@test size(x) == (1, 7680)  # 1 channel, 60 seconds @ 128Hz resolution
+@test size(y) == (1, 2)    # 2 labels, one for each 30 second segment
+
+# Additionally, we can draw down the entire batch at once
+batch, new_state = iterate_batch(batches, RNG)
+X, Y = materialize_batch(batch)
+@test size(X) == (1, 7680, 3)  # we now draw 3 samples and concatenate along the 3rd dimension
+@test size(Y) == (1, 2, 3)
+
+# Note that we can continue to draw as many batches as we like by repeatedly passing the
+# new_state back to iterate_batch, much like Julia's Iterator interface
+# https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-iteration
+# In this way, we can dynamically allocate and load batches of data depending on the needs
+# of the model and infrastructure resources that are available.
+batch, new_state = iterate_batch(batches, RNG)
+X2, Y2 = materialize_batch(batch)
+@test X2 != X
+@test Y2 != Y
+
+###
+### Batcher
+###
+
+# Now that we have the mechanism for generating pseudo-random batches, we need to have these
+# dynamically allocated and loaded in parallel across julia processes. This will enable us
+# to scale our training process across a cluster.
+
+# Let's start by adding a few processes and loading the necessary packages
+addprocs(4)
+@everywhere begin
+    using Pkg
+    using OndaBatches
+end
+
+# Note that the "Batch Manager" cannot be assigned to the primary "Manager" node (1) because XXX
+batch_workers = workers()
+batch_manager = popfirst!(batch_workers)
+
+# A Batcher governs the allocation of batch processing on a distributed environment.
+# We'll provide the RandomBatcher defined above.
+batcher = Batcher(batch_manager, batch_workers, batches; start=false, state=RNG)
+
+# First let's check the initialised batcher hasn't started
+@test get_status(batcher) == :stopped
+@test !isready(batcher.channel)
+
+# Now let's start the batcher - but given we already used the RandomBatcher previously, we
+# need to provide the correct new_state
+# XXX: why does it work if we use state=RNG above? why do we need to do this twice?
+start!(batcher, RNG)
+
+# It should now be running and ready to allocated batches across nodes
+@test get_status(batcher) == :running
+@test !isready(batcher.channel)
+
+# X, Y are the collection of training / evaluation data as above
+# Similarly, we can keep sampling from this by repeatedly passing in the new_state
+(X3, Y3), new_state = take!(batcher, RNG)
+@test size(X3) == size(X)
+@test size(Y3) == size(Y)
+
+# A more convenient way to do this is to just take! from the batcher.channel, which obviates
+# the need for passing around the new_state
+((X4, Y4), new_state), old_state = take!(batcher.channel)
```

Comment:
It's not documented that it can/should be used this way but I've seen this used on beacon-internal code? is it sanctioned? seems safer & more convenient than carrying around a `state`?

Reply:
yeah this is no good 🙅 for the reasons above: it's the responsibility of the caller to handle iteration state.  the other reason is that `take!(::Batcher, state)` handles error propagation, avoiding deadlock, etc. all that good stuff (which is surprisingly tricky to get right).

if a user wants to just have a channel of batches, they are absolutely free to call [`start_batching`](https://github.com/beacon-biosignals/OndaBatches.jl/blob/fa7550a5f25cda2979017f8aa15ffc11d019758f/src/batch_services.jl#L131-L233) with the workers they wanna use; the idea of the `Batcher` is to encapsulate all that trickiness of running this stuff in a distributed way

Reply:
gotcha! I might show you where I saw this pattern internally and you might shed light on what I'm misunderstanding